### PR TITLE
[CDE-943]-Table component that has click action, whenever gets an empty resultset throws an error in jquery.dataTables file.

### PR DESCRIPTION
**Pull request by denisprotopopov**
-fixed

### CODE REVIEW

File: core-js/src/main/javascript/lib/dataTables/js/jquery.dataTables.js [hunk 1]
```diff
@@ -589,11 +589,13 @@
...
-					cell.columnVisible,
-					cell.column
-				];
+				if (cell !== null && cell !== undefined) {
+					return [
+						cell.row,
+						cell.columnVisible,
+						cell.column
+					];
+				}
```
Comment (afrjorge) [inline core-js/src/main/javascript/lib/dataTables/js/jquery.dataTables.js]: Customizing third party plugins is a bad idea, is updating to a newer version an option?
Or is there an alternative way to tackle this issue?

Comment (denisprotopopov) [reply core-js/src/main/javascript/lib/dataTables/js/jquery.dataTables.js]: @afrjorge I have checked the latest version. It has the same issue.

File: pentaho-js/src/main/javascript/cdf/components/TableComponent.js [hunk 2]
```diff
@@ -600,6 +600,7 @@ define([
             target = target.closest('td');
           }
           var position = myself.dataTable.fnGetPosition(target.get(0));
+          if (position == null) { return; }
```
Comment (afrjorge) [inline pentaho-js/src/main/javascript/cdf/components/TableComponent.js]: Alternatively you can do some validations before initializing the position variable, such as:
```
if(!myself.dataTable.api(true).cell( target.get(0) ).index()) {
  console.warn("Click on invalid data detected");
  return;
}
var position = myself.dataTable.fnGetPosition(target.get(0));
```

**Comment by denisprotopopov** [comment]
@mbatchelor @afrjorge please review

**Pull request merged**